### PR TITLE
implement method to get data by user ID

### PR DESCRIPTION
**Pull request by igorskh**
This pull request implements getting user data from `https://www.duolingo.com/2017-06-30/users/{self.user_data.id}`.

This URL is currently used by Duolingo web app, also it is used in this library in the `get_daily_xp_progress`. However, it seems reasonable to have it implemented in a generic way to give users opportunity to request different data.

The reason for introducing fields is that the response is generally quite large, 760 bytes in my case and takes about 3 seconds to receive. One option is to get data on initialisation, but it is dynamic, for example after purchasing an item it has to be requested again.

It should also help users to solve #91 while keeping library code purely for an API interaction, sample code for this:
```python
import duolingo

lingo = duolingo.Duolingo('I2UpM5so', '<password>')

user_data_resp = lingo._get_data_by_user_id(['shopItems'])

search_for = 'streak_freeze'
shop_items = list(filter(
    lambda item: item["id"] == search_for,
    user_data_resp['shopItems']
))

item = shop_items[0] if shop_items else None
if item and item["quantity"]:
    print(f'Current quantity of {search_for}: {item["quantity"]}')

```

Sample output:
```bash
Current quantity of streak_freeze: 2
```

### CODE REVIEW

File: duolingo.py [hunk 1]
```diff
@@ -120,6 +120,15 @@ def _check_login(self):
...
         return resp.status_code == 200
 
+    def get_user_url_by_id(self, fields=None):
+    if fields is None:
+        fields = []
+        url = 'https://www.duolingo.com/2017-06-30/users/{}'.format(self.user_data.id)
+        fields_params = requests.utils.requote_uri(','.join(fields))
+        if fields_params:
+            url += '?fields={}'.format(fields_params)
+        return url
```
Comment (joshcoales) [inline duolingo.py:130]: This all needs un-indenting, or it'll only run when fields is None.  I guess github's suggestion editor auto-indented the rest, whoops.

**Review by joshcoales — COMMENTED** [review]
Yeah, I think that would make sense, it would make things neater for `get_daily_xp_progress()` to use `_get_data_by_user_id()`

File: duolingo.py (hunk 1 above)
Comment (igorskh) [reply duolingo.py:130]: oh, didn't notice. I've fixed in one of the new commits

**Comment by joshcoales** [comment]
Looks good!

**Review by joshcoales — APPROVED** [review]

**Pull request merged**